### DISCUSSION TITLE
Promotions are not working

### DISCUSSION THREAD

**Discussion by Sihana-Shybin** [vendurehq/vendure#2392] [Q&A]
It would be helpful to address the queries regarding the following promotions:
1. Promotion for a particular user group: an order created for a user who is associated with a user group not getting the discount from the promotion set to the user group.
![image](https://github.com/vendure-ecommerce/vendure/assets/94037777/15616b82-e2b5-4f71-bda6-e3b006590385)
2. Promotion with a condition Buy X products, get Y products free not working with the following action.
![image](https://github.com/vendure-ecommerce/vendure/assets/94037777/e03f9a61-7a8c-4a89-b35c-797e46cdcd2f)

I would greatly appreciate it if someone could provide guidance on these queries:-)

**Comment by michaelbromley** [comment]
Hi,

in the first instance (customer group), could it be that the assigned customer group has been deleted? This would explain the "empty" entry in the select input.

in the second instance, could you please provide exact steps to reproduce the issue, including what you think should happen, and what happens instead?

**Reply by Sihana-Shybin** [reply]
Hi,

For the first instance, the customer group is not deleted, and when we click on that green color, then only that customer group is visible.
Even if the customer group is added with the existing active customer, the promotion is not reflected in the order.

In the second instance, I have created a promotion with BuyXGetY. There I have mentioned Buy 1 product and Get 1 product and added the corresponding product as well in the condition field. Then in the Action field, I marked the above option from the dropdown. But when I added the product to the order, the free product was not getting added to the order.

Please check the above use case and help me to sort it out.

**Reply by michaelbromley** [reply]
Do your promotions have any other conditions such as date range or coupon code which may be preventing application of the discount?

**Reply by Sihana-Shybin** [reply]
I have marked only the start date and it is valid as well. A coupon code is also there and I have applied the same coupon in the order. Even after that, it is not reflected in the order.
